pkg/binary/envoy/debug/pii.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2019 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package debug
+
+import (
+	"crypto/sha256"
+	"fmt"
+
+	"bitbucket.org/creachadair/shell"
+)
+
+// ProcessLogs process logs by filtering it using format and containsPII, Hash the resulting array of logs and return the final array.
+// an empty array of strings and an error instance will be returned in the event of an error
+func ProcessLogs(logs []string, format string, containsPII map[string]bool) ([]string, error) {
+	// filter the valid logs according to the format str
+	fieldNames, ok := shell.Split(format)
+	if !ok {
+		return []string{}, fmt.Errorf("error in splitting format string: %s", format)
+	}
+
+	out := make([]string, 0, len(logs))
+	for _, log := range logs {
+		fieldValues, ok := shell.Split(log)
+		if !ok {
+			return []string{}, fmt.Errorf("error in splitting log: %s", log)
```

Comment:
Here we probably don't want to bail on processing everything. Instead we probably just want to skip this one log line and carry on with the others (so just log and `continue` rather than `return`).